dev/com.ibm.ws.concurrent.mp.cdi/src/com/ibm/ws/concurrent/mp/cdi/ConcurrencyCDIExtension.java

```diff
@@ -61,20 +61,113 @@
 
     private static final TraceComponent tc = Tr.register(ConcurrencyCDIExtension.class);
 
+    // This instance is used as a marker for when no configured is specified for a String[]. The reference is compared; its content does not matter.
+    private static final String[] UNSPECIFIED_ARRAY = new String[] {};
+
     // insertion order of the following two data structures must match
     private final ArrayList<String> injectionPointNames = new ArrayList<>();
-    private final Map<String, Annotation> instanceNameToConfig = new LinkedHashMap<>();
+    private final Map<String, Object> instances = new LinkedHashMap<>();
```

Comment:
I'm concerned about memory leaks with this global map. Really the only purpose of this map is to collect info during the `processInjectionPoint/processProducer` phase and then hand off the information to `registerBeans`. So can we clear the map after we are done processing the information in `registerBeans`?

Reply:
I have added some code to clear the lists here.  However, I think another concern is what happens to executors that are pre-built but never make it to the registerBeans method due to a failure in between.